plugins/source/gitlab/codegen/recipes/projects.go

```diff
@@ -0,0 +1,24 @@
+package recipes
+
+import (
+	"github.com/xanzy/go-gitlab"
+)
+
+func Projects() []*Resource {
+	resources := []*Resource{
+		{
+			Service:    "projects",
+			SubService: "projects",
+			PKColumns:  []string{"base_url", "id"},
+			Struct:     &gitlab.Project{},
+			Relations:  []string{"ProjectsReleases()"},
+		},
+		{
+			Service:    "projects",
+			SubService: "projects_releases",
+			Struct:     &gitlab.Release{},
```

Comment:
```suggestion
			Struct:     &gitlab.Release{},
			PKColumns: []string{"base_url", "project_id", "tag_name"},
```

Reply:
Can't `tag_name` as that can be null because a release could be done based on a commit ID